internal/prometheusbpint/modules.go

```diff
@@ -0,0 +1,31 @@
+package prometheusbpint
+
+import (
+	"runtime/debug"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var goModules = promauto.With(GlobalRegistry).NewGaugeVec(prometheus.GaugeOpts{
+	Name: "baseplate_go_modules",
+	Help: "Export the version information for included Go modules, and whether the module is the 'main' module or a 'dependency'.  Always 1",
+}, []string{"go_module", "module_role", "replaced", "module_version"})
+
+func RecordModuleVersions(info *debug.BuildInfo) {
+	record := func(role string, mod *debug.Module) {
+		goModules.With(prometheus.Labels{
+			"go_module":      mod.Path,
+			"module_role":    role,
+			"replaced":       strconv.FormatBool(mod.Replace != nil),
```

Comment:
🔕 I think overall we want to avoid using single word labels, as the whole label indices are global so two unrelated metrics emitting with the same label (e.g. conflicted labels) will cause searching by label slower. I'd suggest use `module_replaced` instead.

Reply:
hmm, makes it a not-strict backport but I can make the change to v2 as well.